### PR TITLE
Added group components to the behavior package

### PR DESCRIPTION
**Pull request by casals**
### Contains
New group components. This relates directly with PR #3717 and with [WildAnimalsMadness PR #1](https://github.com/Terasology/WildAnimalsMadness/pull/1).

### Objective
The new Group components were created as part of the WildAnimalsMadness module. After testing the different grouping scenarios they are ready to me moved into the behavior package as originally intended.

### Changes

- New component: `GroupTag`
- New component: `GroupMind`

### CODE REVIEW

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

File: engine/src/main/java/org/terasology/logic/behavior/GroupMindComponent.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+package org.terasology.logic.behavior;
+
+import org.terasology.entitySystem.Component;
+import org.terasology.entitySystem.entity.EntityRef;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class GroupMindComponent implements Component {
```
Comment (eviltak) [inline engine/src/main/java/org/terasology/logic/behavior/GroupMindComponent.java]: Is it possible to document this component now, or will it be done later?

Comment (casals) [reply engine/src/main/java/org/terasology/logic/behavior/GroupMindComponent.java]: Sorry, didn't realize it was missing - fixed with additional commit.

**Comment by GooeyHub** [comment]
Hooray Jenkins reported success with all tests good!

**Pull request merged**